core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -802,6 +807,17 @@ public List<Transaction> getTransactions() {
         return transactions == null ? null : Collections.unmodifiableList(transactions);
     }
 
+    /**
+     * Gets the transaction at the given index.
+     *
+     * @param index index of the transaction to get
+     * @return transaction
+     * @throws IndexOutOfBoundsException if the given index is out of bounds
+     */
+    public Transaction transaction(int index) {
+        return transactions.get(index);
```

Comment:
Needs null-check

Reply:
Added, though it would throw NPE anyway.